specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2020-01-01/Marketplace.json

```diff
@@ -0,0 +1,552 @@
+              "readOnly": true
+            }
+          }
+        }
+      }
+    },
+    "PrivateStoreList": {
+      "description": "Describes the json payload for the list of available private stores (between zero and one, inclusive)",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateStoreProperties"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next set of PrivateStore list results if there are any.",
+          "type": "string"
+        }
+      }
+    },
+    "PrivateStoreProperties": {
+      "description": "Describes the json payload on whether or not the private store is enabled for a given tenant",
+      "type": "object",
+      "properties": {
+        "availability": {
+          "description": "Indicates private store availability",
+          "type": "string",
+          "enum": [
+            "enabled",
+            "disabled"
+          ],
+          "x-ms-enum": {
+            "name": "Availability",
+            "modelAsString": true
+          }
+        },
+        "name": {
+          "description": "Private Store name",
+          "type": "string"
+        }
+      }
+    },
+    "OfferListResponse": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Offer"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next set of offer list results if there are any.",
+          "type": "string"
+        }
+      }
+    },
+    "Offer": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Offers unique id",
+          "type": "string",
+          "readOnly": true
+        },
+        "displayName": {
+          "description": "It will be displayed prominently in the marketplace",
+          "type": "string",
+          "readOnly": true
+        },
+        "publisherDisplayName": {
+          "description": "Publisher name that will be displayed prominently in the marketplace",
+          "type": "string",
+          "readOnly": true
+        },
+        "summary": {
+          "description": "Brief purpose or function of offer",
+          "type": "string",
+          "readOnly": true
+        },
+        "longSummary": {
+          "description": "Purpose or function of offer",
+          "type": "string",
+          "readOnly": true
+        },
+        "description": {
+          "description": "Description of offer",
+          "type": "string",
+          "readOnly": true
+        },
+        "eTag": {
+          "description": "Identifier for purposes of race condition",
+          "type": "string",
+          "readOnly": true
+        },
+        "plans": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Plan"
+          }
```

Comment:
Probably I confused..

Use readOnly for only those properties that can't be used in PUT and are server generated or derived.

Also, both PUT and GET must have the same definition that they accept in body (PUT) and in response (PUT and GET)
I see eTag is writable in PUT, but here it is marked as readonly.

Also, you may consider using x-ms-mutability ["create", "update", "read"] if you want to scope the field to PUT, PATCH & GET respectively.  (you can use a combination of those)

Reply:
done